### PR TITLE
3088: FBS handler update

### PR DESCRIPTION
**Pull request by jeppekroghitk**
[3088](https://leantime.itkdev.dk/#/tickets/showTicket/3088)

Updated endpoints to use newest operations. Implemented new way of getting patron data, and adjusted formats of data objects.

### CODE REVIEW

File: web/modules/custom/os2forms_fbs_handler/src/Client/FBS.php [hunk 1]
```diff
@@ -228,6 +266,10 @@ private function request(string $uri, array|string $data, string $method = Reque
 
     $response = $this->client->request($method, $url, $options);
 
+    if ($response->getStatusCode() === 204) {
+      return TRUE;
```
Comment (cableman) [inline web/modules/custom/os2forms_fbs_handler/src/Client/FBS.php:270]: Need to change function return docs to say " Json response from FBS or TRUE on create response" or something like that.

File: web/modules/custom/os2forms_fbs_handler/src/Client/Model/Guardian.php [hunk 2]
```diff
@@ -25,7 +25,7 @@ public function __construct(
    */
   public function toArray(): array {
     return [
-      'cprNumber' => $this->cpr,
+      'personIdentifier' => $this->cpr,
```
Comment (cableman) [inline web/modules/custom/os2forms_fbs_handler/src/Client/Model/Guardian.php:28]: Optional: When calling FBS and creating the child, in the last part of this pr, CPR is named `personId`, should this be named the same in the guardian?

**Review by cableman — APPROVED** [review]
Approved with optional comments

File: web/modules/custom/os2forms_fbs_handler/src/Client/Model/Guardian.php (hunk 2 above)
Comment (jeppekroghitk) [reply web/modules/custom/os2forms_fbs_handler/src/Client/Model/Guardian.php:28]: The name of the key is actually personIdentifier and not personId for this operation. There is quite a mismatch between key-naming across the different endpoints.

**Pull request merged**